### ISSUE TITLE
NG SERVE is missing modules when recompiling.

### ISSUE THREAD

**Issue by juninhocruzg3** [angular/angular#31561]
# 🐞 NG SERVE is missing modules when recompiling.

### Command (mark with an `x`)

```
- [ ] new
- [ ] build
- [X] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

 No, It occurs only in ANGULAR CLI 8 generated app

### Description

 Generating an application with Angular CLI 8 and using ng serve, when editing any code, modules are lost when recompiling.

## 🔬 Minimal Reproduction

```
$ ng new portal-sib --enable-ivy --style=scss
$ ng add @angular/material
(import any module - e.g. MatButtonModule - and include html code)
$ ng serve
(Everything works)
(Edit any code and save. After re-compile, mat-button no longer works)
```

I also tried to import material manually but I get the same error.
I also tried to work without Angular Material but I can't use ngFor in my own module because it works only on first ng serve and miss CommonModule on re-compile.

## 🔥 Exception or Error

No error is displayed when the problem occurs with the Angular Material. But when I'm using ngFor in my own module, the following error appears:

<pre><code>

ERROR Error: Template error: Can't bind to 'ngForOf' since it isn't a known property of 'li'.
    at createUnknownPropertyError (core.js:15130)
    at elementPropertyInternal (core.js:14996)
    at Module.ɵɵproperty (core.js:16581)
    at CarouselBannersComponent_Template (carousel-banners.component.html:6)
    at executeTemplate (core.js:14545)
    at checkView (core.js:15969)
    at componentRefresh (core.js:15727)
    at refreshChildComponents (core.js:14222)
    at refreshDescendantViews (core.js:14128)
    at checkView (core.js:15970)
</code></pre>

## 🌍 Your Environment
<pre><code>

Angular CLI: 8.2.0-next.0
Node: 10.14.1
OS: linux x64
Angular: 8.2.0-next.1
... common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.802.0-next.0
@angular-devkit/build-angular     0.802.0-next.0
@angular-devkit/build-optimizer   0.802.0-next.0
@angular-devkit/build-webpack     0.802.0-next.0
@angular-devkit/core              8.2.0-next.0
@angular-devkit/schematics        8.2.0-next.0
@angular/cli                      8.2.0-next.0
@ngtools/webpack                  8.2.0-next.0
@schematics/angular               8.2.0-next.0
@schematics/update                0.802.0-next.0
rxjs                              6.4.0
typescript                        3.4.5
webpack                           4.35.3

</code></pre>

**Anything else relevant?**

This is my first bug report. I hope I have given you enough and clear information. :D

**Comment by varadero** [comment]
Probably the same problem (most likely related to Ivy) as described in https://github.com/angular/angular/issues/31560

**Comment by gkalpak** [comment]
Thx for reporting :+1:
I am going to close this, since it does indeed look similar to #31560 (and thus #31512).

Please, follow #31512 and open a new issue if your problem is not fixed by the fix for #31512.
